### DISCUSSION TITLE
Adding information to a file without overwriting the previous one

### DISCUSSION THREAD

**Discussion by ThundePro** [octokit/octokit.net#2852] [Q&A]
Hello i need a problem with save text in file. I want to add information to the end of a file without changing the previous content. Is this possible without first downloading the contents of this file?

**Comment by kfcampbell** [comment]
This is not possible, sorry. The [repository file API](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents) requires that the entire file content be passed when updating a file.
